a981d3738014339cd0fae1e7f5aa9429222ee03d..ab9f7d4ee689c29784edb20e577c3d82488114f0

From da42cf905d07613ad5243ddee16486369025eb6e Mon Sep 17 00:00:00 2001
From: Henri Verbeet <hverbeet@codeweavers.com>
Date: Wed, 8 May 2019 17:51:36 +0430
Subject: [PATCH] wined3d: Store the glEnableWINE and glDisableWINE pointers in
 the wined3d_gl_info structure.

Signed-off-by: Henri Verbeet <hverbeet@codeweavers.com>
Signed-off-by: Alexandre Julliard <julliard@winehq.org>
---

 dlls/wined3d/context.c         | 2 +-
 dlls/wined3d/directx.c         | 4 ++--
 dlls/wined3d/gl_compat.c       | 8 ++++----
 dlls/wined3d/state.c           | 4 ++--
 dlls/wined3d/wined3d_gl.h      | 2 --
 dlls/wined3d/wined3d_private.h | 3 +++
 6 files changed, 12 insertions(+), 11 deletions(-)


diff --git a/dlls/wined3d/context.c b/dlls/wined3d/context.c
index 1e357e70661..4b7fd7e7890 100644
--- a/dlls/wined3d/context.c
+++ b/dlls/wined3d/context.c
@@ -2052,5 +2052,5 @@ static void SetupForBlit(const struct wined3d_device *device, struct wined3d_con
     checkGLcall("glDisable GL_DEPTH_TEST");
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_ZENABLE));
-    glDisableWINE(GL_FOG);
+    gl_info->p_glDisableWINE(GL_FOG);
     checkGLcall("glDisable GL_FOG");
     context_invalidate_state(context, STATE_RENDER(WINED3D_RS_FOGENABLE));
diff --git a/dlls/wined3d/directx.c b/dlls/wined3d/directx.c
index 4b4209a86a1..d9d4ef4dd1a 100644
--- a/dlls/wined3d/directx.c
+++ b/dlls/wined3d/directx.c
@@ -5861,6 +5861,6 @@ static BOOL wined3d_adapter_init(struct wined3d_adapter *adapter, UINT ordinal)
 #endif
 
-    glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
-    glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
+    gl_info->p_glEnableWINE = gl_info->gl_ops.gl.p_glEnable;
+    gl_info->p_glDisableWINE = gl_info->gl_ops.gl.p_glDisable;
 
     if (!AllocateLocallyUniqueId(&adapter->luid))
diff --git a/dlls/wined3d/gl_compat.c b/dlls/wined3d/gl_compat.c
index 7e6b5101bc4..a2fb09c39e1 100644
--- a/dlls/wined3d/gl_compat.c
+++ b/dlls/wined3d/gl_compat.c
@@ -438,8 +438,8 @@ void install_gl_compat_wrapper(struct wined3d_gl_info *gl_info, enum wined3d_gl_
             old_fogcoord_glFogfv = gl_info->gl_ops.gl.p_glFogfv;
             gl_info->gl_ops.gl.p_glFogfv = wine_glFogfv;
-            old_fogcoord_glEnable = glEnableWINE;
-            glEnableWINE = wine_glEnable;
-            old_fogcoord_glDisable = glDisableWINE;
-            glDisableWINE = wine_glDisable;
+            old_fogcoord_glEnable = gl_info->p_glEnableWINE;
+            gl_info->p_glEnableWINE = wine_glEnable;
+            old_fogcoord_glDisable = gl_info->p_glDisableWINE;
+            gl_info->p_glDisableWINE = wine_glDisable;
 
             old_fogcoord_glVertex4f = gl_info->gl_ops.gl.p_glVertex4f;
diff --git a/dlls/wined3d/state.c b/dlls/wined3d/state.c
index 91d64171f0e..343fe0c0a0e 100644
--- a/dlls/wined3d/state.c
+++ b/dlls/wined3d/state.c
@@ -1031,5 +1031,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     {
         /* No fog? Disable it, and we're done :-) */
-        glDisableWINE(GL_FOG);
+        gl_info->p_glDisableWINE(GL_FOG);
         checkGLcall("glDisable GL_FOG");
         return;
@@ -1168,5 +1168,5 @@ void state_fog_fragpart(struct wined3d_context *context, const struct wined3d_st
     }
 
-    glEnableWINE(GL_FOG);
+    gl_info->p_glEnableWINE(GL_FOG);
     checkGLcall("glEnable GL_FOG");
     if (new_source != context->fog_source || fogstart == fogend)
diff --git a/dlls/wined3d/wined3d_gl.h b/dlls/wined3d/wined3d_gl.h
index 437f62f793a..d82df6ff626 100644
--- a/dlls/wined3d/wined3d_gl.h
+++ b/dlls/wined3d/wined3d_gl.h
@@ -30,6 +30,4 @@
 #define GL_COMPRESSED_LUMINANCE_ALPHA_3DC_ATI 0x8837  /* not in the gl spec */
 
-void (WINE_GLAPI *glDisableWINE)(GLenum cap) DECLSPEC_HIDDEN;
-void (WINE_GLAPI *glEnableWINE)(GLenum cap) DECLSPEC_HIDDEN;
 
 /* OpenGL extensions. */
diff --git a/dlls/wined3d/wined3d_private.h b/dlls/wined3d/wined3d_private.h
index 53ef63e0326..88a8c9fc317 100644
--- a/dlls/wined3d/wined3d_private.h
+++ b/dlls/wined3d/wined3d_private.h
@@ -1749,4 +1749,7 @@ struct wined3d_gl_info
 
     struct wined3d_format *formats;
+
+    void (WINE_GLAPI *p_glDisableWINE)(GLenum cap);
+    void (WINE_GLAPI *p_glEnableWINE)(GLenum cap);
 };
 
--
